### PR TITLE
Return the error message while failed to search with DiskANN

### PR DESCRIPTION
**Pull request by yah01**
/kind improvement

### CODE REVIEW

**Comment by yah01** [comment]
related https://github.com/milvus-io/milvus/issues/26419

**Review by liliu-z — APPROVED** [review]
/lgtm

File: src/index/diskann/diskann.cc [hunk 1]
```diff
@@ -502,18 +503,19 @@ expected<DataSetPtr>
...
-        return Status::invalid_metric_type;
+        return expected<DataSetPtr>::Err(Status::invalid_metric_type, "unsupported metric type");
     }
     auto max_search_list_size = std::max(kSearchListSizeMaxValue, search_conf.k.value() * 10);
     if (search_conf.search_list_size.value() > max_search_list_size ||
         search_conf.search_list_size.value() < search_conf.k.value()) {
         LOG_KNOWHERE_ERROR_ << "search_list_size should be in range: [topk, max(200, topk * 10)]";
-        return Status::out_of_range_in_json;
+        return expected<DataSetPtr>::Err(Status::out_of_range_in_json,
+                                         "search_list_size should be in range: [topk, max(200, topk * 10)]");
```
Comment (jiaoew1991) [inline src/index/diskann/diskann.cc:518]: The input value and the specific value of topk should also be included in the error message.

File: src/index/diskann/diskann.cc [hunk 2]
```diff
@@ -576,16 +578,16 @@ expected<DataSetPtr>
...
     if (!is_prepared_.load() || !pq_flash_index_) {
         LOG_KNOWHERE_ERROR_ << "Failed to load diskann.";
-        return Status::empty_index;
+        return expected<DataSetPtr>::Err(Status::empty_index, "index not loaded");
     }
 
     auto search_conf = static_cast<const DiskANNConfig&>(cfg);
     if (!CheckMetric(search_conf.metric_type.value())) {
-        return Status::invalid_metric_type;
+        return expected<DataSetPtr>::Err(Status::invalid_metric_type, "unknown metric type");
```
Comment (jiaoew1991) [inline src/index/diskann/diskann.cc:586]: The input `metric_type` should be included in error message

**Pull request merged**